version: '3.8'
services:
  dynamodb-local:
    command: "-jar DynamoDBLocal.jar -sharedDb -dbPath ./data"
    image: "amazon/dynamodb-local:latest"
    container_name: dynamodb-local
    ports:
      - "8000:8000"
    volumes:
      - "./docker/dynamodb:/home/dynamodblocal/data"
    working_dir: /home/dynamodblocal
  pulsar:
    container_name: pulsar
    image: 'apachepulsar/pulsar:2.10.0'
    ports:
      - '8080:8080'
      - '6650:6650'
    expose:
      - 8080
      - 6650
    environment:
      PULSAR_MEM: " -Xms512m -Xmx512m -XX:MaxDirectMemorySize=1g"
    command: |
      /bin/bash -c "bin/apply-config-from-env.py conf/standalone.conf && bin/pulsar standalone"
  client-jaeger:
   image: jaegertracing/all-in-one:1.22.0
   container_name: client-jaeger
   ports:
      - "14268:14268"   # jaeger-collector HTTP server (tracer provider)
      - "16686:16686"   # HTTP server (browser UI)